core/native/src/main/scala/scodec/bits/ByteVectorCrossPlatform.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2013, Scodec
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package scodec.bits
+
+import scala.scalanative.unsafe.Ptr
+import scala.scalanative.unsafe.Zone
+import scala.scalanative.unsafe.alloc
+
+private[bits] trait ByteVectorCrossPlatform { self: ByteVector =>
+  import ByteVector._
+
+  def copyToBytePtr(dest: Ptr[Byte], start: Long): Unit =
+    copyToBytePtr(dest, start, 0, size)
+
+  def copyToBytePtr(dest: Ptr[Byte], start: Long, offset: Long, size: Long): Unit = {
+    var i = 0L
+    while (i < size) {
+      dest(start + i) = self(offset + i)
+      i += 1
+    }
+  }
+
+  def toBytePtr(implicit zone: Zone): Ptr[Byte] = {
```

Comment:
Bikeshed "`BytePtr`"? Just `Ptr` or `NativePtr` or `PtrByte`?

Reply:
`toPtr` is my preference assuming there's not a precedence set by Scala.js

Reply:
On Scala.js I used `Uint8Array` (the name of the JS type) and `JSArrayBuffer` to distinguish from the scala one.